specification/iotcentral/data-plane/Microsoft.IoTCentral/preview/1.2-preview/iotcentral.json

```diff
@@ -0,0 +1,5643 @@
+          },
+          {
+            "$ref": "#/parameters/ComponentName"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Success",
+            "schema": {
+              "$ref": "#/definitions/DeviceProperties"
+            }
+          },
+          "default": {
+            "description": "Error",
+            "schema": {
+              "$ref": "#/definitions/Error"
+            }
+          }
+        }
+      },
+      "put": {
+        "operationId": "Devices_ReplaceComponentProperties",
+        "summary": "Replace device properties for a specific component",
+        "x-ms-examples": {
+          "Update properties on components": {
+            "$ref": "./examples/devices_properties_components_set.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/Version"
+          },
+          {
+            "$ref": "#/parameters/DeviceId"
+          },
+          {
+            "$ref": "#/parameters/ComponentName"
+          },
+          {
+            "in": "body",
+            "name": "body",
+            "description": "Device properties.",
+            "schema": {
+              "$ref": "#/definitions/DeviceProperties"
+            },
+            "required": true
+          }
+        ],
+        "responses": {
+          "202": {
+            "description": "Success",
+            "schema": {
+              "$ref": "#/definitions/DeviceProperties"
+            }
+          },
+          "default": {
+            "description": "Error",
+            "schema": {
+              "$ref": "#/definitions/Error"
+            }
+          }
+        }
+      },
+      "patch": {
+        "operationId": "Devices_UpdateComponentProperties",
+        "summary": "Update device properties for a specific component via patch",
+        "x-ms-examples": {
+          "Update properties on components": {
+            "$ref": "./examples/devices_properties_components_set.json"
+          }
+        },
+        "consumes": [
+          "application/json",
+          "application/json-patch+json"
+        ],
+        "parameters": [
+          {
+            "$ref": "#/parameters/Version"
+          },
+          {
+            "$ref": "#/parameters/DeviceId"
+          },
+          {
+            "$ref": "#/parameters/ComponentName"
+          },
+          {
+            "in": "body",
+            "name": "body",
+            "description": "Device properties patch.",
+            "schema": {
+              "type": "object"
+            },
+            "required": true
+          }
+        ],
+        "responses": {
+          "202": {
+            "description": "Success",
+            "schema": {
+              "$ref": "#/definitions/DeviceProperties"
+            }
+          },
+          "default": {
+            "description": "Error",
+            "schema": {
+              "$ref": "#/definitions/Error"
+            }
```

Comment:
Operations with a 202 response should specify `x-ms-long-running-operation: true`.